packages/internal/src/callback.js

```diff
@@ -0,0 +1,122 @@
+// @ts-check
+import { E } from '@endo/far';
+
+/**
+ * @template {(...args: unknown[]) => any} I
+ * @typedef {import('./types').Callback<I>} Callback
+ */
+
+/**
+ * @template {(...args: unknown[]) => any} I
+ * @typedef {import('./types').NearCallback<I>} NearCallback
+ */
+
+/**
+ * Synchronously call a callback.
+ *
+ * @template {(...args: unknown[]) => any} I
+ * @param {NearCallback<I>} callback
+ * @param {Parameters<I>} args
+ * @returns {ReturnType<I>}
+ */
+export const call = (callback, ...args) => {
+  const { target, method, bound } = callback;
+  if (method === undefined) {
+    return target(...bound, ...args);
+  }
+  return target[method](...bound, ...args);
+};
+
+/**
+ * Eventual send to a callback.
+ *
+ * @template {(...args: unknown[]) => any} I
+ * @param {Callback<I>} callback
+ * @param {Parameters<I>} args
+ * @returns {Promise<Awaited<ReturnType<I>>>}
+ */
+export const esend = (callback, ...args) => {
+  const { target, method, bound } = callback;
+  if (method === undefined) {
+    return E(target)(...bound, ...args);
+  }
+  return E(target)[method](...bound, ...args);
+};
+
+/**
+ * Create a callback from a near function.
+ *
+ * @template {(...args: unknown[]) => any} I
+ * @template {(...args: [...B, ...Parameters<I>]) => ReturnType<I>} [T=I]
+ * @template {unknown[]} [B=[]]
+ * @param {T} target
+ * @param {B} bound
+ * @returns {NearCallback<I>}
+ */
+export const func = (target, ...bound) => {
```

Comment:
suggestion:
```suggestion
export const makeFunctionCallback = (target, ...bound) => {
```
(and likewise for later exports, e.g. `makeFarFunctionCallback`/`makeMethodCallback`/`makeFarMethodCallback`)